packages/webpack/src/runtime/runtimeBuilder.js

```diff
@@ -16,191 +60,335 @@ module.exports = {
    *   options
    */
   runtimeBuilder({ options }) {
-    const runtimeOptions = {
-      scuttleGlobalThis: options.scuttleGlobalThis,
-      lockdown: options.lockdown,
+    /**
+     * Generates the preamble that caches selected globals to protect runtime
+     * from scuttling. The set of cached globals is limited to ones known to be
+     * used by runtime modules for the sake of bundle size.
+     *
+     * @returns {string}
+     */
+    function getDefensiveCodingPreamble() {
+      const globals = [
+        'location', // AutoPublicPathRuntimeModule.js
+        'setTimeout', // LoadScriptRuntimeModule.js
+        'clearTimeout', // LoadScriptRuntimeModule.js
+        'document', // LoadScriptRuntimeModule.js, AutoPublicPathRuntimeModule.js
+        'trustedTypes', // GetTrustedTypesPolicyRuntimeModule.js
+        'self',
+      ]
+      return `var ${globals.map((g) => `${g} = globalThis.${g}`).join(',')};
+const LOCKDOWN_SHIMS = [];`
```

Comment:
This doesn't appear to capture any variables from its function context, so maybe worth pulling out?